test/functional/p2p_blockfilters.py

```diff
@@ -280,6 +303,220 @@ def run_test(self):
+        # Calculate remaining amount (minus small fee)
+        fee = Decimal("0.0001")
+        remaining = Decimal(str(unspent['amount'])) - amount_locked - fee
+
+        # Create the transaction
+        lock_tx = CTransaction()
+        lock_tx.vin = inputs
+
+        # Add change output if there's remaining amount
+        if remaining > 0:
+            change_addr = node.getnewaddress()
+            change_script = bytes.fromhex(node.validateaddress(change_addr)['scriptPubKey'])
+            lock_tx.vout = [CTxOut(int(remaining * COIN), change_script)]
+
+        # Add OP_RETURN output with the locked amount
+        lock_tx.vout.append(CTxOut(int(amount_locked * COIN), CScript([OP_RETURN, b""])))
+
+        lock_tx.nVersion = 3
+        lock_tx.nType = 8  # asset lock type
+        lock_tx.vExtraPayload = lockTx_payload.serialize()
+
+        # Sign the transaction
+        tx_hex = lock_tx.serialize().hex()
+        signed_tx = node.signrawtransactionwithwallet(tx_hex)
+        return tx_from_hex(signed_tx["hex"]), pubkey
+
+    def test_special_transactions_in_filters(self):
+        """Test that special transactions are included in block filters.
+
+        Note: This functional test only covers AssetLockTx transactions because:
+        - ProRegTx requires 1000 DASH collateral and masternode setup
+        - ProUpServTx/ProUpRegTx/ProUpRevTx require existing masternodes
+        - All special transaction types are comprehensively tested in unit tests
+          (see src/test/blockfilter_tests.cpp)
+
+        The test verifies filter functionality by comparing filter sizes between
+        blocks with and without special transactions, as GCS filter decoding
+        is not readily available in Python test framework.
+        """
+        self.log.info("Test that special transactions are included in block filters")
+
+        # Restart node 0 with blockfilterindex enabled
+        self.restart_node(0, extra_args=["-blockfilterindex", "-peerblockfilters"])
+
+        # Test AssetLockTx transactions
+        self.test_assetlock_basic()
+        self.test_assetlock_multiple_outputs()
+
+        self.log.info("Special transaction block filter test completed successfully")
+
+    def test_assetlock_basic(self):
+        """Test that AssetLockTx credit outputs are included in block filters"""
+        self.log.info("Testing AssetLockTx in block filters...")
+
+        node = self.nodes[0]
+
+        # Create an AssetLockTx
+        amount_to_lock = Decimal("0.1")
+        lock_tx, pubkey = self.create_simple_assetlock(amount_to_lock)
+
+        # Send the transaction
+        txid = node.sendrawtransaction(lock_tx.serialize().hex())
+
+        # Verify it's recognized as an AssetLockTx
+        tx_details = node.getrawtransaction(txid, True)
+        assert tx_details.get('type') == 8, f"Expected type 8 (AssetLockTx), got {tx_details.get('type')}"
+        # Note: vExtraPayload may not appear in RPC output but the transaction is still valid
```

Comment:
nit: could be a bit more strightforward
```suggestion
        assert 'assetLockTx' in tx_details, "Transaction should have assetLockTx field"
```